tools/get_dicom_files.pl

```diff
@@ -0,0 +1,292 @@
+my $dbh                = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+my $tarchiveLibraryDir = NeuroDB::DBI::getConfigSetting(\$dbh, 'tarchiveLibraryDir');
+
+#----------------------------------------------------------#
+# Create the directory where DICOM files will be extracted #
+#----------------------------------------------------------#
+my $tmpExtractDir = sprintf("%s/%s.%d", $tmpExtractBaseDir, basename($0), $$);
+mkdir $tmpExtractDir or die "Cannot create directory $tmpExtractDir: $!\n";
+
+#------------------------------------------------------------#
+# Get the list of tarchives for which the patient names      #
+# match the list of supplied patterns (or all tarchives if   #
+# no patterns were specified)                                #
+#------------------------------------------------------------#
+my $query = "SELECT c.PSCID, s.Visit_label, t.DateAcquired, t.ArchiveLocation, t.TarchiveID, t.PatientName "
+          . "FROM tarchive t "
+          . "JOIN session s ON (t.SessionID=s.ID) "
+          . "JOIN candidate c ON (c.CandID=s.CandID) ";
+if(@patientNames) {
+    my @where = map { "t.PatientName LIKE ?" } @patientNames;
+    $query .= sprintf("WHERE %s", join("OR", @where));
+}
+my $sth = $dbh->prepare($query);
+
+if(@patientNames) {
+    for(my $p=0; $p<@patientNames; $p++) {
+        $sth->bind_param($p+1, $patientNames[$p]);
+    }
+}
+$sth->execute();
+
+#---------------------------------------------------------#
+# Process each tarchive and extract the DICOMs associated #
+# to the scan types of interest                           #
+#---------------------------------------------------------#
+my @outDirs = ();
+foreach my $tarchiveRowRef (@{ $sth->fetchall_arrayref }) {
+    my($pscid, $visitLabel, $dateAcquired, $archiveLocation, $tarchiveId) = @$tarchiveRowRef;
+    
+    # Extract only the .tar.gz archive from the main archive (ignore the 
+    # meta data and log file)
+    print "Extracting compressed archive contained in $tmpExtractDir...";
+    system("tar xf $tarchiveLibraryDir/$archiveLocation -C $tmpExtractDir --wildcards \\*.tar.gz") == 0 
+      or die "Extraction of compressed archive from $tarchiveLibraryDir/$archiveLocation in directory $tmpExtractDir failed: $?";
+    print "done\n";
+    
+    # Fetch all the MINC files created out of the DICOM archive whose 
+    # acquisition protocolswhose acquisition protocols match the scan types
+    # of interest
+    $query = "SELECT f.File, tf.FileName "
+           . "FROM files f "
+           . "JOIN mri_scan_type mst ON (mst.ID=f.AcquisitionProtocolID) "
+           . "JOIN tarchive_series ts ON (f.SeriesUID=ts.SeriesUID AND f.EchoTime*1000=ts.EchoTime) "
+           . "JOIN tarchive_files tf USING (TarchiveSeriesID) "
+           . "WHERE f.TarchiveSource = ? ";
+    my @where = map { "mst.Scan_type LIKE ?" } @scanTypes;
+    $query .= sprintf(" AND (%s)", join("OR", @where)) if @where;
```

Comment:
will end up with the same SQL error in the where part as mentioned in line 159.

Reply:
Re-Arghhh! Fixed.